datapreprocess/ray/tokenize_shuffle.py

```diff
@@ -111,10 +113,16 @@ def dl_parse_s3(data, creds=None):
         client = boto3.client("s3")
     out_dicts = []
     for path in data["path"]:
-        bucket, key = parse_s3_path(path)
-        json_lines = download_to_memory_with_progress(bucket, key).decode().splitlines()
-        jsons = [json.loads(x) for x in json_lines]
-        out_dicts += jsons
+        # Download from s3 if s3 path
+        if path.startswith("s3://"):
```

Comment:
do we wanna use fsspec here?

Reply:
I kept this as is since there is a custom download function with progress bar in the s3 case. Not particularly inclined to either case here, unless @Vaishaal has points to make re: performance.